File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    stack = [words[0]]
    for i in range(1, len(words)):
        if words[i] not in stack:
            if words[i-1][-1] != words[i][0]:
                return [i % n + 1, i // n + 1]
            else:
                stack.append(words[i])
        else:
            return [i % n + 1, i // n + 1]
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        Set<String> used = new HashSet<>();
        used.add(words[0]);

        for (int i = 1; i < words.length; i++) {
            String prev = words[i - 1];
            String curr = words[i];

            if (used.contains(curr) || curr.charAt(0) != prev.charAt(prev.length() - 1)) {
                int who = (i % n) + 1;
                int when = (i / n) + 1;
                return new int[]{who, when};
            }

            used.add(curr);
        }

        return new int[]{0, 0};
    }
}
